src/teleop/jetson/jetson_teleop.py

```diff
@@ -0,0 +1,138 @@
+#!/usr/bin/env python3
+# Node for teleop-related callback functions
+
+from math import copysign
+from enum import IntEnum
+import rospy as ros
+from sensor_msgs.msg import Joy, JointState
+from geometry_msgs.msg import Twist
+from mrover.msg import Chassis, RAOpenLoopCmd, HandCmd
+
+
+def quadratic(val):
+    return copysign(val**2, val)
+
+
+def deadzone(magnitude, threshold):
+    temp_mag = abs(magnitude)
+    if temp_mag <= threshold:
+        temp_mag = 0
+    else:
+        temp_mag = (temp_mag - threshold) / (1 - threshold)
+
+    return copysign(temp_mag, magnitude)
+
+
+class Drive:
+    class LogitechAxes(IntEnum):
+        left_right = 0
+        forward_back = 1
+        twist = 2
+        dampen = 3
+        pan = 4
+        tilt = 5
+
+    # Constants for diff drive
+    TRACK_RADIUS = 10.0  # meter
+    WHEEL_RADIUS = 0.5  # meter
+
+    drive_vel_pub = ros.Publisher("/drive_cmd_wheels", Chassis, queue_size=100)
+
+    # TODO: Reimplement Dampen Switch
+    def teleop_drive_callback(self, msg):
+
+        # teleop_twist_joy angular message is maxed at 0.5 regardless of
+        # turbo mode, multiply by 2 to make range [-1,1]
+        v = msg.linear.x
+        omega = msg.angular.z * 2
+
+        # Transform into L & R wheel angular velocities using diff drive kinematics
+        omega_l = (v - omega * self.TRACK_RADIUS / 2.0) / self.WHEEL_RADIUS
+        omega_r = (v + omega * self.TRACK_RADIUS / 2.0) / self.WHEEL_RADIUS
+
+        command = Chassis()
+        command.omega_l = omega_l
+        command.omega_r = omega_r
+
+        self.drive_vel_pub.publish(command)
+
+
+class ArmControl:
+    class XboxMappings(IntEnum):
+        left_js_x = 0
+        left_js_y = 1
+        left_trigger = 6
+        right_trigger = 7
+        right_js_x = 2
+        right_js_y = 3
+        right_bumper = 5
+        left_bumper = 4
+        d_pad_up = 12
+        d_pad_down = 13
+        d_pad_right = 14
+        d_pad_left = 15
+        a = 0
+        b = 1
+        x = 2
+        y = 3
+
+    # RA Joint Publishers
+    joint_a_pub = ros.Publisher("/ra/open_loop/joint_a", JointState, queue_size=100)
+    joint_b_pub = ros.Publisher("/ra/open_loop/joint_b", JointState, queue_size=100)
+    joint_c_pub = ros.Publisher("/ra/open_loop/joint_c", JointState, queue_size=100)
+    joint_d_pub = ros.Publisher("/ra/open_loop/joint_d", JointState, queue_size=100)
+    joint_e_pub = ros.Publisher("/ra/open_loop/joint_e", JointState, queue_size=100)
+    joint_f_pub = ros.Publisher("/ra/open_loop/joint_f", JointState, queue_size=100)
+
+    # Hand Publishers
+    finger_pub = ros.Publisher("/hand/open_loop/finger", JointState, queue_size=100)
+    grip_pub = ros.Publisher("/hand/open_loop/grip", JointState, queue_size=100)
+
+    def ra_control_callback(self, msg):
+
+        print(self.XboxMappings.left_js_x)
+
+        joint_a = JointState()
+        joint_a.velocity.append(quadratic(deadzone(msg.axes[self.XboxMappings.left_js_x], 0.15)))
```

Comment:
I suggest using the rosparam server to allow users to easily change controls for various joints. It would be useful for configuring at least inversion. For dealing with mappings of controls, it might be harder because it depends on if you're using the analog sticks vs. the buttons so I won't push that hard on that but it is definitely worth exploring.

e.g.
![image](https://user-images.githubusercontent.com/71603173/187779326-c31b463e-719d-49b2-af16-24021dc931da.png)

Reply:
Rosparam is something I haven't really explored, would the values specified in this yaml file be accessible via rospy.get_param('teleop/ra_controls/joint_a/multiplier') for example? Do I need to add this config file to teleop.launch or anything?

Reply:
Will most likely do this tomorrow at some point

Reply:
Thanks